tests/components/mqtt/test_discovery.py

```diff
@@ -63,6 +63,58 @@
 )
 
 
+@pytest.fixture
+def mqtt_data_flow_calls() -> list[MqttServiceInfo]:
+    """Return list to capture MQTT data data flow calls."""
+    return []
+
+
+@pytest.fixture
+async def mock_mqtt_flow(
+    hass: HomeAssistant, mqtt_data_flow_calls: list[MqttServiceInfo]
+) -> config_entries.ConfigFlow:
+    """Test fixure for mqtt integration flow.
+
+    The topic is used as a unique ID.
+    The component test domain used is: `comp`.
+
+    Creates an entry if does not exist.
+    Updates an entry if it exists, and there is an updated payload.
+    Removess an entry if an empty payload is received.
+    """
+
+    class TestFlow(config_entries.ConfigFlow):
+        """Test flow."""
+
+        async def async_step_mqtt(self, discovery_info: MqttServiceInfo) -> FlowResult:
+            """Test mqtt step."""
+            await asyncio.sleep(0)
+            mqtt_data_flow_calls.append(discovery_info)
+            # Abort a flow if there is an update a d update the existing entry or
+            # remove the entry is the payload is empty.
+            if entry := self.hass.config_entries.async_entry_for_domain_unique_id(
+                "comp", discovery_info.topic
+            ):
+                if discovery_info.payload:
+                    hass.config_entries.async_update_entry(
+                        entry,
+                        data={
+                            "name": discovery_info.topic,
+                            "payload": discovery_info.payload,
+                        },
+                    )
+                else:
+                    await hass.config_entries.async_remove(entry.entry_id)
+                raise AbortFlow("already_configured")
+            await self.async_set_unique_id(discovery_info.topic)
+            return self.async_create_entry(
+                title="Test",
+                data={"name": discovery_info.topic, "payload": discovery_info.payload},
+            )
+
+    return TestFlow
```

Comment:
This change does not seem to be related to the re-discovery functionality introduced by the PR?
Can it please be moved to a separate PR?

Reply:
It is related, it was a sub class before, now it is a fixture.

Reply:
Moved removing the entry into the test